api/lib/infrastructure/events/EventDispatcher.js

```diff
@@ -0,0 +1,30 @@
+class EventDispatcher {
+  constructor() {
+    this._subscriptions = [];
+  }
+
+  subscribe(event, eventHandler) {
+    this._preventDuplicateSubscription(event, eventHandler);
+    this._subscriptions.push([event.prototype.constructor, eventHandler]);
+  }
+
+  _preventDuplicateSubscription(event, eventHandler) {
+    const duplicateSubscription = this._subscriptions.find(([e, eh]) => e == event && eh == eventHandler);
+    if (duplicateSubscription != null) {
+      throw new Error('Cannot subscribe twice to a given event with the same handler');
+    }
+  }
+
+  async dispatch(domainTransaction, dispatchedEvent) {
+    const subscriptions = this._subscriptions.filter(([event, _]) => {
+      return dispatchedEvent instanceof event;
+    });
+
+    for (const subscription of subscriptions) {
+      const returnedEvent = await subscription[1]({ domainTransaction, event:dispatchedEvent });
```

Comment:
En utilisant un objet on aurait plus de "magic index".
```suggestion
      const returnedEvent = await subscription.eventHandler({ domainTransaction, event: dispatchedEvent });
```